src/main/java/org/sagebionetworks/bridge/spring/controllers/CRCController.java

```diff
@@ -189,47 +208,48 @@ public StatusMessage updateParticipant(@PathVariable String userId) {
 
     @PutMapping("/v1/cuimc/observations")
     public ResponseEntity<StatusMessage> postObservation() {
-        httpBasicAuthentication();
+        App app = httpBasicAuthentication();
         
         IParser parser = FHIR_CONTEXT.newJsonParser();
         JsonNode data = parseJson(JsonNode.class);
         Observation observation = parser.parseResource(Observation.class, data.toString());
         
         String userId = findUserId(observation.getIdentifier());
         
-        int status = writeReportAndUpdateState(userId, data, OBSERVATION_REPORT, AccountStates.TESTS_AVAILABLE);
+        int status = writeReportAndUpdateState(app, userId, data, OBSERVATION_REPORT, AccountStates.TESTS_AVAILABLE);
         if (status == 200) {
             return ResponseEntity.ok(new StatusMessage("Observation updated."));
         }
         return ResponseEntity.created(URI.create("/v1/cuimc/observations/" + userId))
                 .body(new StatusMessage("Observation created."));        
     }
     
-    void createLabOrder(StudyParticipant participant) {
+    void createLabOrder(Account account) {
         // Call external partner here and submit the patient record
         // this will trigger workflow at Columbia.
-        /*
-        Patient patient = createPatient(participant);
+        Patient patient = createPatient(account);
         IParser parser = FHIR_CONTEXT.newJsonParser();
         String json = parser.encodeResourceToString(patient);
         
         try {
-            HttpResponse response = Request.Post("<url unknown>")
-                    .bodyString(json, APPLICATION_JSON)
-                    .execute()
-                    .returnResponse();
+            HttpResponse response = post(json);
             if (response.getStatusLine().getStatusCode() != 200 && 
                 response.getStatusLine().getStatusCode() != 201) {
-                LOG.error("Error submitting patient record to CUIMC for user " + participant.getId());
-                throw new BridgeServiceException("The server encountered an error");
+                LOG.error(EXT_ERROR + account.getId());
+                throw new ServiceUnavailableException(UNAVAILABLE_ERROR);
```

Comment:
I don't have strong feelings about this because I haven't seen their interface. Hopefully we catch any issues that would generate a 4xx series of errors, but I'll update this.

Reply:
From the end user's perspective, there's nothing they should be able to do in our API that would cause their API to return a 3xx or 4xx error (if these happen, it is a programming error on the part of the Bridge server). 5xx errors are not distinguishable from the end user's perspective (whether it came from us or Columbia doesn't matter). I would suggest:

3xx and 4xx - we convert to 500 internal server error, and log it as error
5xx - we pass these through because 503 is temporary, but log it as error